src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,42 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Encapsulates a {@link Task} that has a deadline to meet.
+ */
+public class Deadline extends Task {
+    private final LocalDate date;
+
+    public Deadline(String content, LocalDate date) {
+        super(content);
+        this.date = date;
+    }
+
+    /**
+     * returns a string representation of the deadline to be seen by users.
+     *
+     * @return a string representation of the deadline to be seen by users.
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + date.format(DateTimeFormatter.BASIC_ISO_DATE) + ")";
+    }
+
+    /**
+     * returns a string representation of the deadline to be used for data storage.
+     *
+     * @return a string representation of the deadline to be used for data storage.
+     */
+    @Override
+    public String toData() {
+        String isFinishedData;
+        if (super.finished) {
+            isFinishedData = "1";
+        } else {
+            isFinishedData = "0";
+        }
+        return "D:" + isFinishedData + ":" + super.content + ":" + date.format(DateTimeFormatter.ISO_DATE);
+    }
```

Comment:
Some comments about this:
1) I think isFinishedData could be renamed to something else since it sounds like a boolean variable. Maybe try "doneStatus" or "finishedStatus"?
2) Since you are using strings to show boolean states maybe you can change the code to look something like this. I feel that simply assigning the isFinishedData to a number does not tell readers much immediately.:
```
@Override
    public String toData() {
        String finishedStatus;
        String completedStatus = "1";
        String incompleteStatus = "0";
        if (super.finished) {
            doneStatus = completedStatus;
        } else {
           doneStatus = incompleteStatus;
        }
        return "D:" + finishedStatus + ":" + super.content + ":" + date.format(DateTimeFormatter.ISO_DATE);
    }
```